src/Castle.Core/DynamicProxy/Generators/Emitters/MethodEmitter.cs

```diff
@@ -160,6 +160,117 @@ private void DefineParameters(ParameterInfo[] parameters)
 				{
 					parameterBuilder.SetCustomAttribute(attribute.Builder);
 				}
+
+				// If a parameter has a default value, that default value needs to be replicated.
+				// Default values as reported by `ParameterInfo.[Raw]DefaultValue` have two possible origins:
+				//
+				// 1. A `[DecimalConstant]` or `[DateTimeConstant]` custom attribute attached to the parameter.
+				//    Attribute-based default values have already been copied above.
+				//
+				// 2. A `Constant` metadata table entry whose parent is the parameter.
+				//    Constant-based default values need more work. We can detect this case by checking
+				//    whether the `ParameterAttributes.HasDefault` flag is set. (NB: This is not the same
+				//    as querying `ParameterInfo.HasDefault`, which would also return true for case (1)!)
+				if ((parameter.Attributes & ParameterAttributes.HasDefault) != 0)
+				{
+					CopyDefaultValueConstant(from: parameter, to: parameterBuilder);
+				}
+			}
+		}
+
+		private void CopyDefaultValueConstant(ParameterInfo from, ParameterBuilder to)
+		{
+			Debug.Assert(from != null);
+			Debug.Assert(to != null);
+			Debug.Assert((from.Attributes & ParameterAttributes.HasDefault) != 0);
+
+			object defaultValue;
+			try
+			{
+				defaultValue = from.DefaultValue;
+			}
+			catch (FormatException) when (from.ParameterType == typeof(DateTime))
```

Comment:
`.DefaultValue` may also throw a `FormatException` for (closed generic) enum parameter types; see https://github.com/dotnet/corefx/issues/29570. Another `catch` dedicated for this issue might be necessary here.

Reply:
Done. 🏁